### PR TITLE
Add openSUSE:Factory key

### PR DESCRIPTION
**Pull request by dcermak**
We need to add the signkey of openSUSE:Factory to the gpg keys as well, so that we can pull registry.opensuse.org/opensuse/nginx for the php tests. And for that, we need a newer podman…

Requires #221

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -203,9 +214,43 @@ jobs:
...
+          Rq/lROEF5lx4SxB838JU4ezxD++BJXfBTE8JZmlGscXv74y9nCtSOZza8KOKj8ou
+          WRl739FMnx9jRd7HHj3TIyymoveODnZ7f3IElyyFsjBW3XuQ9XfpZrIkwHuaZV5M
+          6q2h+hgWNQ==
+          =nMh8
+          -----END PGP PUBLIC KEY BLOCK-----
+          EOF
 
       - name: Run tox job
-        run: sudo --preserve-env tox -e ${{ matrix.toxenv }} -- -n auto
+        run: sudo --preserve-env tox -e ${{ matrix.toxenv }} -- -n auto --reruns 3
```
Comment (dirkmueller) [inline .github/workflows/ci.yaml:253]: do we really want to do this? do we have such a flakey testsuite?

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -144,11 +144,22 @@ jobs:
...
         if: ${{ matrix.container_runtime == 'PODMAN' }}
         run: |
-          sudo apt update && sudo apt install podman
+          sudo mkdir -p /etc/apt/keyrings
+          curl -fsSL https://download.opensuse.org/repositories/devel:kubic:libcontainers:unstable/xUbuntu_$(lsb_release -rs)/Release.key \
+            | gpg --dearmor \
+            | sudo tee /etc/apt/keyrings/devel_kubic_libcontainers_unstable.gpg > /dev/null
+          echo \
+            "deb [arch=$(dpkg --print-architecture) signed-by=/etc/apt/keyrings/devel_kubic_libcontainers_unstable.gpg]\
+              https://download.opensuse.org/repositories/devel:kubic:libcontainers:unstable/xUbuntu_$(lsb_release -rs)/ /" \
```
Comment (dirkmueller) [inline .github/workflows/ci.yaml:157]:
```suggestion
              https://download-repositories.opensuse.org/repositories/devel:kubic:libcontainers:unstable/xUbuntu_$(lsb_release -rs)/ /" \
```

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -144,11 +144,22 @@ jobs:
         uses: actions/setup-python@v4
         with:
           python-version: '3.x'
+
       - name: Install necessary dependencies
         if: ${{ matrix.container_runtime == 'PODMAN' }}
         run: |
-          sudo apt update && sudo apt install podman
+          sudo mkdir -p /etc/apt/keyrings
+          curl -fsSL https://download.opensuse.org/repositories/devel:kubic:libcontainers:unstable/xUbuntu_$(lsb_release -rs)/Release.key \
```
Comment (dirkmueller) [inline .github/workflows/ci.yaml:152]:
```suggestion
          curl -fsSL https://download-repositories.opensuse.org/repositories/devel:kubic:libcontainers:unstable/xUbuntu_$(lsb_release -rs)/Release.key \
```

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (dcermak) [reply .github/workflows/ci.yaml:253]: I have seen so many tests fail with connection failures to download.suse.com, this _might_ help.

**Comment by dcermak** [comment]
This is still not going to work as the opensuse/nginx image is signed with the key `D754694F9AB48CE9` which is whatever :shrug: but most certainly **not** the key from `openSUSE:Factory`

**Comment by dirkmueller** [comment]
> This is still not going to work as the opensuse/nginx image is signed with the key `D754694F9AB48CE9` which is whatever shrug but most certainly **not** the key from `openSUSE:Factory`

This is the container signing key, aka the right one. I've updated the PR to fix this.

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (dirkmueller) [reply .github/workflows/ci.yaml]: I have reverted that. we should do that conciously in a separete PR if it really is still a problem

**Comment by dirkmueller** [comment]
PHP tests have passed

**Pull request merged**